tests/vmi_configuration_test.go

```diff
@@ -2349,4 +2349,80 @@ var _ = Describe("Configurations", func() {
 			table.Entry("[test_id:3808] Should be rejected when using ide", "ide", "IDE bus is not supported"),
 		)
 	})
+
+	Context("Custom PCI Adresses with slots and function", func() {
+
+		var vmi *v1.VirtualMachineInstance
+
+		containerImage := tests.ContainerDiskFor(tests.ContainerDiskCirros)
+
+		checkPciAddress := func(vmi *v1.VirtualMachineInstance, expectedPciAddress string, prompt string) {
+			err := tests.CheckForTextExpecter(vmi, []expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: prompt},
+				&expect.BSnd{S: "grep DEVNAME /sys/bus/pci/devices/" + expectedPciAddress + "/*/block/vda/uevent|awk -F= '{ print $2 }'\n"},
+				&expect.BExp{R: "vda"},
+			}, 15)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		It("should configure custom pci address across all slots", func() {
+			pciSlotsAddr := make([]string, 24)
+			for i := range pciSlotsAddr {
+				pciSlotsAddr[i] = fmt.Sprintf("%x", i+2)
+				if len(pciSlotsAddr[i]) == 1 {
+					pciSlotsAddr[i] = "0" + pciSlotsAddr[i]
+				}
+			}
+			pciAddr := make([]string, len(pciSlotsAddr))
+			for i, v := range pciSlotsAddr {
+				pciAddr[i] = fmt.Sprintf("0000:00:%v.0", v)
+			}
+
+			for _, pci := range pciAddr {
```

Comment:
Would it be possible to start 1 VM with 24 disks instead of 24 VMs? That would make the test a lot quicker and avoid timeouts on slow hardware.

Reply:
Broke it down to 2 VMs now with half of the PCI addresses
VMs seems to be unreachable with more than 20 Devices